### PR TITLE
Perf Improvements 3

### PR DESCRIPTION
**Pull request by developit**
This improves performance by another 5-10% compared to master (5.2.2):
Screen Shot 2022-08-21 at 6 36 54 PM

It also corrects the type definitions, reduces size by ~50% and removes pretty-printing entirely from the default bundle. It can still be accessed via `preact-render-to-string/jsx`.

**Because of the exports/options change, I'm marking this as a major release.**

### CODE REVIEW

**Review by JoviDeCroock — APPROVED** [review]
Let's merge this after we do one more release with the _diff and diffed cahnges

**Comment by developit** [comment]
@JoviDeCroock good call, definitely worth getting those landed on 5.x.

File: src/util.js [hunk 1]
```diff
@@ -1,17 +1,15 @@
 // DOM properties that should NOT have "px" added when numeric
 export const IS_NON_DIMENSIONAL = /acit|ex(?:s|g|n|p|$)|rph|grid|ows|mnc|ntw|ine[ch]|zoo|^ord|^--/i;
-export const VOID_ELEMENTS = /^(area|base|br|col|embed|hr|img|input|link|meta|param|source|track|wbr)$/;
+export const VOID_ELEMENTS = /^(?:area|base|br|col|embed|hr|img|input|link|meta|param|source|track|wbr)$/;
```
Comment (marvinhagemeister) [inline src/util.js:3]: Good catch!

**Review by marvinhagemeister — APPROVED** [review]
I really like the move to split off the `pretty` option from the default render function. This is great 👍

**Review by JoviDeCroock — APPROVED** [review]

File: src/pretty.js [hunk 2]
```diff
@@ -82,17 +120,20 @@ export function _renderToStringPretty(
...
+				let provider = context[cxType.__c];
+				cctx = provider ? provider.props.value : cxType.__;
+			}
 
 			if (
 				!nodeName.prototype ||
 				typeof nodeName.prototype.render !== 'function'
 			) {
-				let cctx = getContext(nodeName, context);
+				// let cctx = getContext(nodeName, context);
```
Comment (marvinhagemeister) [inline src/pretty.js:136]: Nit: Can be dropped

File: src/pretty.js [hunk 3]
```diff
@@ -109,7 +150,7 @@ export function _renderToStringPretty(
 					rendered = nodeName.call(vnode.__c, props, cctx);
 				}
 			} else {
-				let cctx = getContext(nodeName, context);
+				// let cctx = getContext(nodeName, context);
```
Comment (marvinhagemeister) [inline src/pretty.js:153]: Nit: Can be dropped

File: src/index.js [hunk 4]
```diff
@@ -159,79 +99,20 @@ function renderClassComponent(vnode, context) {
 		c.state = c[NEXT_STATE] !== c.state ? c[NEXT_STATE] : c.state;
 	}
 
-	let renderHook = options[RENDER];
+	// let renderHook = options[RENDER];
```
Comment (marvinhagemeister) [inline src/index.js:102]: Nit: can be dropped

File: src/index.js [hunk 5]
```diff
@@ -159,79 +99,20 @@ function renderClassComponent(vnode, context) {
...
 	if (renderHook) renderHook(vnode);
+	// if (options[RENDER]) options[RENDER](vnode);
```
Comment (marvinhagemeister) [inline src/index.js:104]: Nit: Can be dropped

**Review by marvinhagemeister — APPROVED** [review]

**Pull request merged**